doc/HOW_THINGS_WORK.md

```diff
@@ -0,0 +1,225 @@
+The goal of this document is to provide high-level overviews of how
+various systems work inside Chef Server. If you encounter a process
+or feature that isn’t straightforward or took you time to understand,
+consider writing an explanation here.
+
+## Search Indexing
+
+Chef Server performs searches by querying a search index. This index
+is updated on every write to the Chef Server. That is, every time we
+write to oc_erchef's postgresql database, we usually also need to
```

Comment:
recommend: "..database to update an object -- such as a node, data bag item, or role -- we usually also..."